### PR TITLE
Add members, methods and imports to EnumDeclarations

### PR DESCRIPTION
**Pull request by robinmaisch**
Concerning #1445:

- `EnumDeclaration`s now get assigned their class members, just like `RecordDeclaration`s.
- The `ImportResolver` correctly identifies entries and other members of `EnumDeclaration`s that are imported.
- combined the processing of static and non-static `ImportDeclaration`s, separating them afterwards by `isStatic` (a comment indicated that something similar was done earlier and then undone)
- in some cases re-merged the processing of `EnumDeclaration`s and `RecordDeclaration`s and refactored out common parts into separate methods
- added a test and test files for the above changes

### CODE REVIEW

File: cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt [hunk 1]
```diff
@@ -198,112 +198,42 @@ open class DeclarationHandler(lang: JavaLanguageFrontend) :
...
+            // a "this" reference to the outer class, so methods can use a "qualified this"
+            // (OuterClass.this.someFunction()). This is the same as the java compiler does. The
+            // reference
+            // is stored as an implicit field.
+            handleInnerRecord(recordDeclaration)
         }
         return recordDeclaration
     }
 
+    private fun handleInnerRecord(recordDeclaration: RecordDeclaration) {
```
Comment (konradweiss) [inline cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt:220]: Here, we should avoid naming the function `handler` as that is our convention for functions that get an AST object from a parser and return an AST object in our CPG representation. Handlers are also added to a handler map in the init block. Here, I would ask for a descriptive name that does not collide with `handler`, e.g. resolve, process, add, etc.

File: cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt [hunk 2]
```diff
@@ -360,15 +290,108 @@ open class DeclarationHandler(lang: JavaLanguageFrontend) :
...
+            // a "this" reference to the outer class, so methods can use a "qualified this"
+            // (OuterClass.this.someFunction()). This is the same as the java compiler does. The
+            // reference
+            // is stored as an implicit field.
+            handleInnerRecord(enumDeclaration)
+        }
         return enumDeclaration
     }
 
+    private fun <T : TypeDeclaration<T>> handleRecordMembers(
```
Comment (konradweiss) [inline cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt:322]: Same here

File: cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt [hunk 3]
```diff
@@ -360,15 +290,108 @@ open class DeclarationHandler(lang: JavaLanguageFrontend) :
...
+                        recordDeclaration,
+                    )
+                    .implicit(recordDeclaration.name.localName)
+            recordDeclaration.addConstructor(constructorDeclaration)
+            frontend.scopeManager.addDeclaration(constructorDeclaration)
+        }
+        frontend.processAnnotations(recordDeclaration, typeDecl)
+    }
+
+    private fun handleImportDeclarations(recordDeclaration: RecordDeclaration) {
```
Comment (konradweiss) [inline cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt:377]: And here

**Review by konradweiss — CHANGES_REQUESTED** [review]
Overall I like it and would merge it after the names are changed to avoid conflicts with our more restrictive understanding of a `handler`. And it is necessary to resolve the conflicts in the FrontendTest file. Thanks already for you contributions so far!

**Comment by robinmaisch** [comment]
Sorry, my IDE reformatted the code back upon committing. And the commit title was not meant to refer to that issue.

**Comment by konradweiss** [comment]
> Sorry, my IDE reformatted the code back upon committing. And the commit title was not meant to refer to that issue.

It confused me a bit ;)

**Pull request merged**